server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -1032,9 +1032,13 @@ protected String delegatingTo() {
                 return new BlockStoredFieldsReader.BytesFromStringsBlockLoader(name());
             }
 
-            // _ignored_source field will only be present if text field is not stored
-            // and there is no syntheticSourceDelegate
-            if (isSyntheticSource && syntheticSourceDelegate == null) {
+            // _ignored_source field will contain entries for this field if it is not stored
+            // and there is no syntheticSourceDelegate.
+            // See #syntheticSourceSupport().
+            // But if a text field is a multi field it won't have an entry in _ignored_source.
+            // The parent might, but we don't have enough context here to figure this out.
+            // So we bail.
+            if (isSyntheticSource && syntheticSourceDelegate == null && parentField == null) {
```

Comment:
This issue is exposed by new `TextFieldWithParentBlockLoaderTests` so i had to do an inline fix. I think this problem exists for other fields too. I'll create a follow up with more testing and fixes for other fields.